src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java

```diff
@@ -0,0 +1,520 @@
+    }
+
+    private static Stream<String> invalidYearsProvider() {
+        String yearAfterNextYear = Integer.toString(LocalDate.now().getYear() + 2);
+        return Stream.of(" ", "abcd", yearAfterNextYear, "21000");
+    }
+
+    @BeforeEach
+    void setup(WireMockRuntimeInfo runtimeInfo) {
+
+        Environment environment = Mockito.mock(Environment.class);
+        when(environment.readEnv("ALLOWED_ORIGIN")).thenReturn("*");
+        when(environment.readEnv("API_DOMAIN")).thenReturn("localhost");
+        when(environment.readEnv("CUSTOM_DOMAIN_BASE_PATH")).thenReturn("publication-channels");
+        var dataportenBaseUri = URI.create(runtimeInfo.getHttpsBaseUrl());
+        var httpClient = WiremockHttpClient.create();
+        var publicationChannelClient = new DataportenPublicationChannelClient(httpClient, dataportenBaseUri, null);
+
+        this.handlerUnderTest = new SearchJournalByQueryHandler(environment, publicationChannelClient);
+        this.output = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnResultWithSuccessWhenQueryIsIssn() throws IOException, UnprocessableContentException {
+        var year = randomValidYear();
+        var issn = randomIssn();
+        var expectedSearchResult = getExpectedPaginatedSearchResultIssnSearch(year, issn);
+
+        var input = constructRequest(Map.of("year", year, "query", issn));
+
+        this.handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, PaginatedSearchResult.class);
+        var pagesSearchResult = objectMapper.readValue(response.getBody(),
+                new TypeReference<PaginatedSearchResult<JournalResult>>() {
+                });
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(pagesSearchResult.getHits(), containsInAnyOrder(expectedSearchResult.getHits().toArray()));
+    }
+
+    @Test
+    void shouldReturnResultWithSuccessWhenQueryIsPid() throws IOException, UnprocessableContentException {
+        var year = randomValidYear();
+        var pid = UUID.randomUUID().toString();
+        var expectedSearchResult = getExpectedPaginatedSearchResultPidSearch(year, pid);
+
+        var input = constructRequest(Map.of("year", year, "query", pid));
+
+        this.handlerUnderTest.handleRequest(input, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, PaginatedSearchResult.class);
+        var pagesSearchResult = objectMapper.readValue(response.getBody(),
+                new TypeReference<PaginatedSearchResult<JournalResult>>() {
+                });
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(pagesSearchResult.getHits(), containsInAnyOrder(expectedSearchResult.getHits().toArray()));
+    }
+
+    @Test
+    void shouldReturnResultWithSuccessWhenQueryIsName() throws IOException, UnprocessableContentException {
```

Comment:
Big test - extract some of it? Setup method?

Reply:
The setup is kind of clunky. I did some cleaning, but most of it "has" to stay.